### PR TITLE
Remove cosmic-swingset's no-LMDB fallback thing

### LINKED ISSUE #3131: remove no-LMDB fallback from cosmic-swingset
cosmic-swingset currently accomodates platforms which do not support LMDB (https://github.com/Agoric/agoric-sdk/blob/1de9ee6101b0a33b3f0fca853bf02d8a446cbe5a/packages/cosmic-swingset/src/check-lmdb.js) by falling back to the "jsonlines" persistence mode of "swing-store-simple" (which otherwise keeps all of its state in RAM, in one big `Map`). In this mode, `commit()` serializes the entire Map and writes it to a file, and `open()` reads that entire file back into memory. Obviously, this only performs acceptably for toy examples or unit tests, but it's the one we started with, and we didn't remove it after getting LMDB.

I think we added this fallback when we identified LMDB problems on WSL, but I also think we subsequently found a better workaround (tell LMDB to not use sparse files or `mmap()`).

We're adding (#3065, #3122) a feature to the kernelDB that moves large transcript entries out of the key-value store and into linear auxilliary files. The LMDB implementation is the most important one, and we're adding a comparable API to the in-RAM non-persistent flavor of swing-store-simple. But we don't want to implement persistence for the new API, and in fact we intend to get rid of the persistent mode altogether. 

So the task is to change cosmic-swingset to remove this fallback to swing-store-simple. If LMDB doesn't work, cosmic-swingset cannot be run on that platform (since it couldn't be run remotely efficiently there anyways).

_Originally posted by @warner in https://github.com/Agoric/agoric-sdk/pull/3122#discussion_r635461113_

### PR DESCRIPTION
**Pull request by FUDCo**
Closes #3131

### CODE REVIEW

**Review by warner — APPROVED** [review]
looks good to me, although I think we also need to remove `@agoric/swing-store-simple` from the `package.json`

Please wait for @michaelfig 's approval too.

**Comment by FUDCo** [comment]
> looks good to me, although I think we also need to remove `@agoric/swing-store-simple` from the `package.json`
> 
> Please wait for @michaelfig 's approval too.

Yes and yes.

**Review by michaelfig — APPROVED** [review]
Lgtm!

**Pull request merged**
